src/Microsoft.FileFormats/PDB/MSFZFile.cs

```diff
@@ -0,0 +1,408 @@
+
+                uint fileBytesTransferred = _reader.Read(fileOffset, buffer, bufferOffset, transferSize);
+                if (fileBytesTransferred != transferSize)
+                {
+                    // We expect the entire read to be satisfied.
+                    throw new Exception("Internal error in MSFZ reader. The underlying file reader did not read enough data from the file.");
+                }
+
+                count -= transferSize;
+                bufferOffset += transferSize;
+                position += transferSize;
+                totalBytesTransferred += transferSize;
+            }
+
+            return totalBytesTransferred;
+        }
+
+
+        public void Dispose()
+        {
+            if (_reader.DataSource is IDisposable disposable)
+            {
+                disposable.Dispose();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Describes the MSFZ File Header on-disk data structure.
+    /// </summary>
+    public sealed class MSFZFileHeader : TStruct
+    {
+        // 00000000 :  4d 69 63 72 6f 73 6f 66 74 20 4d 53 46 5a 20 43 : Microsoft MSFZ C
+        // 00000010 :  6f 6e 74 61 69 6e 65 72 0d 0a 1a 41 4c 44 00 00 : ontainer...ALD..
+
+        public static readonly byte[] ExpectedSignature =
+        {
+            0x4d, 0x69, 0x63, 0x72, 0x6f, 0x73, 0x6f, 0x66, 0x74, 0x20, 0x4d, 0x53, 0x46, 0x5a, 0x20, 0x43, // : Microsoft MSFZ C
+            0x6f, 0x6e, 0x74, 0x61, 0x69, 0x6e, 0x65, 0x72, 0x0d, 0x0a, 0x1a, 0x41, 0x4c, 0x44, 0x00, 0x00, // : ontainer...ALD..
+        };
+
+        public const ulong VersionV0 = 0;
+
+        public const int SizeOf = 80;
+
+        /// <summary>Identifies this as an MSFZ file.</summary>
+        [ArraySize(32)]
+        public byte[] Signature;
+        /// <summary>specifies the version number of the MSFZ file format</summary>
+        public ulong Version;
+        /// <summary>file offset of the stream directory</summary>
+        public ulong StreamDirOffset;
+        /// <summary>file offset of the chunk table</summary>
+        public ulong ChunkTableOffset;
+        /// <summary>the number of streams stored within this MSFZ file</summary>
+        public uint NumStreams;
+        /// <summary>compression algorithm used for the stream directory</summary>
+        public uint StreamDirCompression;
+        /// <summary>size in bytes of the stream directory when compressed (on disk)</summary>
+        public uint StreamDirSizeCompressed;
+        /// <summary>size in bytes of the stream directory when uncompressed (in memory)</summary>
+        public uint StreamDirSizeUncompressed;
+        /// <summary>number of compressed chunks</summary>
+        public uint NumChunks;
+        /// <summary>size in bytes of the chunk table</summary>
+        public uint ChunkTableSize;
+
+        public bool IsMagicValid
+        {
+            get { return Signature.SequenceEqual(ExpectedSignature); }
+        }
+    }
+
+    public class MSFZConstants
```

Comment:
```suggestion
    internal class MSFZConstants
```
This looked like new public API - if we need to do that it would be good to understand what is consuming the API.